extension/lib/bigcommerce/JwtFactory.js

```diff
@@ -0,0 +1,59 @@
+const jwt = require('jwt-simple')
+const randomBytes = require('randombytes')
+
+/**
+ * @property {string} _clientId
+ * @property {string} _storeHash
+ * @property {string} _clientSecret
+ */
+class JwtFactory {
+  /**
+   * @param {string} clientId
+   * @param {string} storeHash
+   * @param {string} clientSecret
+   */
+  constructor (clientId, storeHash, clientSecret) {
+    this._clientId = clientId
+    this._storeHash = storeHash
+    this._clientSecret = clientSecret
+  }
+
+  /**
+   * @param {string} customerId
+   * @param {string|null} redirectLink
+   */
+  create (customerId, redirectLink = null) {
```

Comment:
This method could be preAuthenticateUrl or preAuthUrl or similar

Reply:
I changed the name to preAuthToken since it returns the token rather than the full URL.